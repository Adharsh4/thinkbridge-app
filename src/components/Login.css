/* .login{
    min-height: 100vh;
    background-color: rgb(245, 243, 239);
} */

.login__container{
    width: 30%;
    height: fit-content;
    margin: 9% auto;
    padding: 2%;
    background-color: white;
    box-shadow: 0 0 5px 2px rgb(207, 193, 193);
}

@media only screen and (max-width: 1050px) and (min-width:900px) {
  .login__container{
    width: 40%;
  }
}

@media only screen and (max-width: 899px) and (min-width:700px) {
  .login__container{
    width: 50%;
  }
}

@media only screen and (max-width: 699px) and (min-width:500px) {
  .login__container{
    width: 60%;
  }
}

@media only screen and (max-width: 499px) and (min-width:400px) {
  .login__container{
    width: 80%;
  }
}

@media only screen and (max-width: 399px) and (min-width:200px) {
  .login__container{
    width: 90%;
  }
}

.form__alingment .form-group{
    margin: 7% 5%;
}


.login__signInButton {
    background: #f0c14b;
    border-radius: 2px;
    width: 60%;
    height: 30px;
    border: 1px solid;
    margin-top: 10px;
    border-color: #a88734 #9c7e31 #846a29;
  }

  .login__RegisterButton{
    background: #f0c14b;
    border-radius: 2px;
    width: 60%;
    height: 30px;
    border: 1px solid;
    margin-top: 10px;
    border-color: #a88734 #9c7e31 #846a29;
  }

  .custom-button{
    background: #f0c14b;
    border-radius: 2px;
    width: 60%;
    height: 30px;
    border: 1px solid;
    margin-top: 10px;
    border-color: #a88734 #9c7e31 #846a29;
  }
  
