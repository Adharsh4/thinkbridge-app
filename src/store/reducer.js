import * as actionTypes from './actionTypes';

const initialState = {
    products: [],
    isLoading: true,
    errorMessage: ""
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case actionTypes.START_FETCHING:
            return {
                ...state,
                isLoading: true,
                errorMessage: ""
            }
        case actionTypes.STOP_FETCHING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.SAVE_PRODUCTS:
            return {
                ...state,
                products: action.products,
                isLoading: false,
                selectedProduct: [],
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADINGTOTRUE:
            return {
                ...state,
                isLoading: true,
            }
        case actionTypes.SET_ERRORMESSAGE:
            return {
                ...state,
                errorMessage: action.message,
                isLoading: false
            }
        default:
            return state;
    }
}

export default reducer;