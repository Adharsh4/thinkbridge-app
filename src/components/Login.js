import React, { useState, useEffect } from 'react';
import './Login.css'
import axios from 'axios';
import { withRouter } from "react-router-dom";
import  validator  from "validator";
import {connect} from 'react-redux';
import * as actions from "../store/actions"
import Loader from "react-loader-spinner"

function Login(props) {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    // const [userToken, setUserToken] = useState(localStorage.getItem("user"));
    const [isClicked, setIsClicked] = useState(false)

    useEffect(() => {
        props.changeIsLoading();
        localStorage.removeItem("email");
    }, [])

    const signinOrRegister = (e, type) => {
        e.preventDefault();
        if (email === "" || password === "" ) {
            const errorMessage = "Please fill all fields"
            props.setErrorMessage(errorMessage)
            alert(errorMessage);
            return;
        }else if(!validator.isEmail(email)){
            const errorMessage = "Please enter valid email"
            props.setErrorMessage(errorMessage)
            alert(errorMessage);
            return;
        } else {
            props.singinOrRegisterUser({email, password, type, history: props.history })
            if(type !== "signin"){
                setEmail("");
                setPassword("");
                setIsClicked(prevState => !prevState);
            }
        }
    }

    const setIsClickedHandler = (e) => {
        e.preventDefault();
        setIsClicked(!isClicked)
    }

    return (
        props.isLoading ? <div className="loading"><h3>Please wait...</h3><Loader type = "Oval" color="#00BFFF" height={30} width={30}/></div> :
            <div className='login__container'>
                {!isClicked ? <h1 className="text-center">Sign-in</h1> : <h1 className="text-center">Sign-up</h1>}

                <form className="form__alingment">
                    <div class="form-group">
                        <input type="email" class="form-control" id="email" placeholder="Enter email" value={email} onChange={e => setEmail(e.target.value)}/>
                        
                    </div>
                    <div class="form-group">
                        <input type="password" class="form-control" id="password" placeholder="Password" value={password} onChange={e => setPassword(e.target.value)} />
                    </div>
                    <div  className="text-center">
                        {!isClicked ? <button  type='submit' onClick={(e) => signinOrRegister(e, "singin")} className='login__signInButton'>Sign In</button> : <button onClick={(e) => signinOrRegister(e, "register")} className='login__RegisterButton'>Register</button>}
                        {!isClicked ? <button className='custom-button' onClick={e => setIsClickedHandler(e)}>Switch to sign up</button> :
                        <button className='custom-button' onClick={e => setIsClickedHandler(e)}>Switch to sign in</button>}
                    </div>
                </form>
            </div>
          
    )

    

}
const mapStateToProps = (state) => {
    return{
        isLoading: state.isLoading
    }
}
 
const mapDispatchToProps = (dispatch) => {
    return{
        changeIsLoading: () => dispatch(actions.changeIsLoading()),
        singinOrRegisterUser : (postObject) => dispatch(actions.singinOrRegisterUser(postObject)),
        setErrorMessage: (message) => dispatch(actions.setErrorMessage(message))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));

