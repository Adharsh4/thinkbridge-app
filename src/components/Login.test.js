
import React from 'react';
import Login from "./Login"
import Enzyme, {shallow, mount} from 'enzyme'
import EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import {BrowserRouter} from 'react-router-dom'
import { createStore, applyMiddleware } from 'redux';
import reducer from "../store/reducer"
import thunk from 'redux-thunk';

Enzyme.configure({adapter: new EnzymeAdapter()})


// const mockStore = createStore(reducer, {
//     products: [],
//     isLoading: false
// });
describe('<Login /> unit test', () => {
    const mockStore = createStore(reducer, applyMiddleware(thunk));
    const getWrapper = () => mount(
      <Provider store={mockStore}>
          <BrowserRouter>
            <Login/>
        </BrowserRouter>
      </Provider>
    );
    const container = getWrapper();

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have an email field', () => {
        expect(container.find('input[type="email"]').length).toEqual(1);
    });

    it('should have an password field', () => {
        expect(container.find('input[type="password"]').length).toEqual(1);
    });

    it('can click the button for singin and register', () => {
      const button = container.find('button.login__signInButton')
      button.simulate('click');
    });

    // it('can click the button for singin and register', () => {
    //   const button = container.find('button.custom-button')
    //   button.simulate('click');
    // });

   

    it('should have proper props for email field', () => {
        const emailField = container.find('input[type="email"]')
        emailField.simulate('change', {target: {value: 'aa@gmail.com'}})
        expect(container.find('#email').props().value).toEqual('aa@gmail.com')
    });

    it('should have proper props for password field', () => {
        const passwordField = container.find('input[type="password"]')
        passwordField.simulate('change', {target: {value: 'aa1998'}})
        expect(container.find('#password').props().value).toEqual('aa1998')
    });


    it('can click the button for singin and register', () => {
        const button = container.find('button.login__signInButton')
        console.log(button.debug())
        button.simulate('click');
      });

      it('can click the button for singin and register', () => {
        const button = container.find('button.custom-button')
        console.log(button.debug())
        button.simulate('click');
      });
      it('can click the button for singin and register', () => {
        const button = container.find('button.custom-button')
        console.log(button.debug())
        button.simulate('click');
      });

      it('should have proper props for email field', () => {
        const emailField = container.find('input[type="email"]')
        emailField.simulate('change', {target: {value: 'aaaa@gmail.com'}})
        // expect(container.find('#email').props().value).toEqual('aa@gmail.com')
    });

    it('should have proper props for password field', () => {
        const passwordField = container.find('input[type="password"]')
        passwordField.simulate('change', {target: {value: 'aaaa1998'}})
        // expect(container.find('#password').props().value).toEqual('aa1998')
    });

      it('can click the button for singin and register', () => {
        const button = container.find('button.login__RegisterButton')
        console.log(button.debug())
        button.simulate('click');
      });

    //   it('can click the button for singin and register', () => {
    //     const button = container.find('button.login__signInButton').at(1)
    //     button.simulate('click');
    //   });

  
  });
