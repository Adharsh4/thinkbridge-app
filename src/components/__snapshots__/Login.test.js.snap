// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> unit test should match the snapshot 1`] = `"<div class=\\"login__container\\"><h1 class=\\"text-center\\">Sign-in</h1><form class=\\"form__alingment\\"><div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" id=\\"email\\" placeholder=\\"Enter email\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"Password\\" value=\\"\\"></div><div class=\\"text-center\\"><button type=\\"submit\\" class=\\"login__signInButton\\">Sign In</button><button class=\\"custom-button\\">Switch to sign up</button></div></form></div>"`;
